simplq/src/components/pages/Admin/index.jsx

```diff
@@ -37,6 +37,22 @@ export default () => {
     }
   };
 
+  const pollAndUpdate = () => {
+    if (queueId) {
+      QueueService.get(queueId)
+        .then((data) => {
+          setItems(data.tokens);
+          setTimeout(pollAndUpdate, 10000);
+        })
+        .catch((err) => {
+          handleApiErrors(err);
+          setTimeout(pollAndUpdate, 10000);
+        });
+    }
+  };
```

Comment:
This is a very interesting problem. I think we can have a single function that takes care of both. Then the function can be called at start using `useEffect` at start and also when the button is clicked.

Use `clearTimeout` to cancel the previous timeout when the function is called again. Can you give this a try? The thing to figure out is where to store the `timeoutID`

Reply:
A recommended way involves using a higher-order component called timeout which has timeout methods and passing in those methods as props to our child component.

An alternative way would be to use the component's state.